packages/date-time/src/components/Calendar/CalendarYear/CalendarYear.base.tsx

```diff
@@ -0,0 +1,458 @@
+
+  public focus(): void {
+    if (this._selectedCellRef) {
+      this._selectedCellRef.focus();
+    } else if (this._currentCellRef) {
+      this._currentCellRef.focus();
+    }
+  }
+  public render(): JSX.Element {
+    const { styles, theme, className, fromYear, toYear } = this.props;
+
+    const classNames = getClassNames(styles, {
+      theme: theme!,
+      className: className
+    });
+
+    let year = fromYear;
+    const cells = [];
+    while (year <= toYear) {
+      cells.push(this._renderCell(year));
+      year++;
+    }
+    return (
+      <FocusZone>
+        <div className={css('ms-DatePicker-optionGrid', classNames.yearGridContainer)} role="grid">
+          <div role="row">{cells}</div>
+        </div>
+      </FocusZone>
+    );
+  }
+  private _renderCell = (year: number): React.ReactNode => {
+    const { styles, theme, minYear, maxYear, onSelectYear, selectedYear } = this.props;
+    const selected = year === selectedYear;
+    const disabled = (minYear !== undefined && year < minYear) || (maxYear !== undefined && year > maxYear);
+    const current = year === new Date().getFullYear();
+    return (
+      <CalendarYearGridCell
+        {...this.props}
+        key={year}
+        year={year}
+        selected={selected}
+        current={current}
+        disabled={disabled}
+        onSelectYear={onSelectYear}
+        ref={selected ? this._onSelectedCellRef : current ? this._onCurrentCellRef : undefined}
+        styles={styles}
+        theme={theme}
+      />
+    );
+  };
+  private _onSelectedCellRef = (ref: CalendarYearGridCell) => {
+    this._selectedCellRef = ref;
+  };
+  private _onCurrentCellRef = (ref: CalendarYearGridCell) => {
+    this._currentCellRef = ref;
+  };
+}
+class CalendarYearNavPrev extends React.Component<ICalendarYearHeaderProps, {}> {
+  public render(): JSX.Element {
+    const { styles, theme, className, navigationIcons, strings, onSelectPrev } = this.props;
+
+    const classNames = getClassNames(styles, {
+      theme: theme!,
+      className: className
+    });
+
+    const iconStrings = navigationIcons || DefaultNavigationIcons;
+    const yearStrings = strings || DefaultCalendarYearStrings;
+    const prevRangeAriaLabel = yearStrings.prevRangeAriaLabel || yearStrings.rangeAriaLabel;
+    const prevAriaLabel = prevRangeAriaLabel
+      ? typeof prevRangeAriaLabel === 'string'
+        ? (prevRangeAriaLabel as string)
+        : (prevRangeAriaLabel as ICalendarYearRangeToString)(this.props)
+      : undefined;
+    const disabled = this.isDisabled;
+
+    return (
+      <button
+        className={css('ms-DatePicker-prevDecade', classNames.yearNavigationButton, {
+          [`ms-DatePicker-prevDecade--disabled ${classNames.disabledStyle}`]: disabled
+        })}
+        onClick={!disabled && onSelectPrev ? this._onSelectPrev : undefined}
+        onKeyDown={!disabled && onSelectPrev ? this._onKeyDown : undefined}
+        type="button"
+        tabIndex={0}
```

Comment:
any particular reason we're putting tabIndex={0} on a button? That should be implied for a button usually

Reply:
Direct copy from other component. Will remove.